src/main/scala/gitbucket/core/controller/api/ApiIssueCommentControllerBase.scala

```diff
@@ -65,9 +65,39 @@ trait ApiIssueCommentControllerBase extends ControllerBase {
   })
 
   /*
-   * v. Edit a comment
-   * https://developer.github.com/v3/issues/comments/#edit-a-comment
+   * v. Update an issue comment
+   * https://developer.github.com/v3/issues/comments/#update-an-issue-comment
    */
+  patch("/api/v3/repos/:owner/:repository/issues/comments/:id")(readableUsersOnly { repository =>
+    val commentId = params("id")
+    getComment(repository.owner, repository.name, commentId) match {
+      case Some(issueComment) =>
+        getIssue(repository.owner, repository.name, issueComment.issueId.toString) match {
+          case Some(issue) =>
+            if (isEditable(issue.userName, issue.repositoryName, issue.openedUserName)) {
+              val body = extractFromJsonBody[CreateAComment].map(_.body)
+              updateCommentByApi(repository, issue, issueComment.commentId.toString, body) match {
+                case Some(_) =>
```

Comment:
You don't have a case for `None` here, won't this trigger a "non-exhaustive match" compiler error?

Reply:
Ah, never mind.  You have a catch-all case
```Scala
case _ => NotFound()
```
on line 98

Reply:
Yeah, there are some compile time warnings...

![image](https://user-images.githubusercontent.com/17608272/91856208-3584b200-eca1-11ea-96d7-ab753fef43a8.png)

Reply:
> Yeah, there are some compile time warnings...

yeah. I didn't put `case _ ` because once you find comment, you are sure there is issue. but caused compile warning...
so I fixed this in https://github.com/gitbucket/gitbucket/pull/2538/commits/69f94f435c1fd476a74499c7e0621a167c743df4